frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx

```diff
@@ -31,9 +31,6 @@ export const getSelectOverrides = (): MantineThemeOverride["components"] => ({
       dropdown: {
         background: "var(--mb-color-background)",
         borderColor: "var(--mb-color-border)",
-        ">div": {
```

Comment:
I had a look at the issue that was caused [here](https://metaboat.slack.com/archives/C057WD5L0JG/p1707929338121809) and I was able to solve it by deleting this, and converting the `SelectDropdown` from a `div` to a `Box` with `display: contents`. This works because we created some unnecessary nesting with the extra div in `SelectDropdown`, which caused the Select items to overflow the `SelectDropdown` div. That's why we needed to have the extra flex, overflow, and maxHeight properties. By using `display: contents`, we remove that nesting from the DOM and we can rely on the default Mantine styles/config to make things work.

I've also manually tested this on a bunch of different selects and it seems to be working as as intended.

Reply:
@WiNloSt the reason I mention this is because this code was actually the thing causing the massive overflow (i.e. the maxHeight being none caused it to go nuts). It's not a perfect solution - the dropdown can still sometimes go off the screen, but it respects the `maxDropdownHeight` in the default theme (`512px`), so essentially it's a cap for now. I think it would be good to figure this out at some other point but I think this is a good start for now

Reply:
TIL: `display: contents`